wallet/src/main/java/io/neow3j/wallet/Account.java

```diff
@@ -75,26 +73,54 @@ public String getLabel() {
         return label;
     }
 
+    public Account setLabel(String label) {
+        this.label = label;
+        return this;
+    }
+
+    public Wallet getWallet() {
+        return this.wallet;
+    }
+
+    /**
+     * Checks if the account is default in the linked wallet.
+     *
+     * @return whether the account is default.
+     */
     public Boolean isDefault() {
-        return isDefault;
+        if (this.wallet == null) return false;
+        return this.wallet.isDefault(this.getScriptHash());
     }
 
     // This method is required by the Wallet but must not be available to the developer because
     // it might bring a wallet into inconsistent state, i.e. having multiple default accounts.
-    void setDefault() {
-        this.isDefault = true;
-    }
+//    void setDefault() {
+//        this.isDefault = true;
+//    }
 
     // This method is required by the Wallet but must not be available to the developer because
     // it might bring a wallet into inconsistent state, i.e. having multiple default accounts.
-    void unsetDefault() {
-        this.isDefault = false;
-    }
+//    void unsetDefault() {
+//        this.isDefault = false;
+//    }
 
     public Boolean isLocked() {
         return isLocked;
     }
 
+    public Account setLocked() {
+        this.isLocked = true;
+        return this;
+    }
+
+    public void unsetLocked() {
+        this.isLocked = false;
+    }
+
+    public void setWallet(Wallet wallet) {
```

Comment:
`setWallet()` should not be public but package-private or protected. The developer should not be able to set the wallet on an account, because this could lead to inconsistent states.